### ISSUE TITLE
Table block: Merge/Unmerge cells

### ISSUE THREAD

**Issue by mapk** [WordPress/gutenberg#15821]
The Table block does not provide a way to merge or unmerge cells. It's a popular feature request for the Table block, let's see if we can make this happen. 😉 

Something similar to this (Google Docs):

![merge](https://user-images.githubusercontent.com/617986/58354214-98906380-7e25-11e9-80db-1a6e7336fbc3.gif)

We can expose this action in the dropdown similarly to how we display the other actions.

![merge-cells](https://user-images.githubusercontent.com/617986/58354583-c4601900-7e26-11e9-9f8c-41eadc6459e1.png)

**Comment by mapk** [comment]
This request was noted here: https://github.com/WordPress/gutenberg/issues/6923 by @helen.

**Comment by talldan** [comment]
Thanks so much for helping define these tasks @mapk! Would definitely be great to get this in. 

Some steps could be taken towards cell merging:
- Support the `colspan` attribute on the underlying HTML (would at least allow merged cells for pasted tables, or those edited in the HTML editor, and needs to be in place anyway)
- Cell navigation and selection needs to be improved (keyboard navigation, multiple cell selection). I started working on keyboard nav, so will hopefully have a PR incoming for that, but haven't had that much time lately to work on things.

**Comment by charlyox** [comment]
Hello here, this feature is really missing indeed, especially in a B2B context ! I can't wait for it. I'm a php / JS dev, I'd be happy to help ! 
Thanks a lot

**Comment by charlyox** [comment]
Would it be a good start to allow html editing with colspan & rowspan attributes ?

**Comment by gziolo** [comment]
The same feature was requested on the Gutenberg plugin forum:
https://wordpress.org/support/topic/table-block-no-cell-merging/

**Comment by dmvvilela** [comment]
So nothing yet?

**Comment by jobthomas** [comment]
Confirming that this would be a great addition to the table block.

**Comment by adamsilverstein** [comment]
Thanks for the reminders, I'll try to pick up work on https://github.com/WordPress/gutenberg/pull/17261 or parallel efforts soon.

**Comment by alexrhino** [comment]
we would definitely need this feature too

**Comment by afercia** [comment]
Reported also on core Trac, see https://core.trac.wordpress.org/ticket/50071

**Comment by ocean90** [comment]
@adamsilverstein Is there something we can help you with? Are there any blockers?

**Comment by sergeiermilov** [comment]
How can I find out the status of this issue?

**Comment by tnchuntic** [comment]
Is it possible to add list block inside table? like innerblock or something.

**Comment by abhishek-pokhriyal** [comment]
This is such a common use case that it's frustrating that it's not already there. I tried to add `rowspan` by converting to HTML but it makes the table block invalid.

Guten Team, please look into this 🙏

**Comment by klostrophobik** [comment]
Mark me down as another request.  I've got a few hundred staff members that are using your tables on our site and would like to be able to do this.

**Comment by sergeiermilov** [comment]
Gutenberg team, are you listening anybody?!

**Comment by jeffpaul** [comment]
@sergeiermilov yes, the team is listening and there's a [work-in-progress PR](https://github.com/WordPress/gutenberg/pull/17261) for this so if this is something or urgent and/or important need for you then please see how you can help on that PR.

**Comment by joelyoder** [comment]
> * Support the `colspan` attribute on the underlying HTML (would at least allow merged cells for pasted tables, or those edited in the HTML editor. This is the basic foundation needed for the cell merging feature to work)

Is there any downside to including colspan support like @talldan mentions above? Assuming it doesn't break the editor, it would be a good stopgap for now. I'm ok with having to add in the colspan manually for now, but not having any support for it becomes frustrating when converting a lot of pages that used that feature.

**Comment by sabernhardt** [comment]
[Trac #52937](https://core.trac.wordpress.org/ticket/52937): another request for simply allowing these attributes in the table block (when using edit as HTML)

**Comment by jdaviescoates** [comment]
I just wanted to do this and so came here and found this issue.  Seems there has been a work-in-progress PR since 2019! Any chance of us getting this anytime soon? Thanks!

**Comment by jeffpaul** [comment]
@jdaviescoates checking in on #18768 is probably best as that's where folks are confirming approach, which will then either need to be updated in #17261 or crafted via a new PR.

**Comment by WhiteX** [comment]
While we wait... what's the best workaround you found? I'm looking for bringing some tables from MS Word to WP, and the lack of the merged cells feature makes it impossible to handle with the default options.

**Comment by joelyoder** [comment]
> While we wait... what's the best workaround you found? I'm looking for bringing some tables from MS Word to WP, and the lack of the merged cells feature makes it impossible to handle with the default options.

The simplest solution I've found in the meantime is just using a table inside of a Classic Block. Anything you could do in TinyMCE before will work in that block.

**Comment by WhiteX** [comment]
Thanks @joelyoder , I tried and it might be a good substitute for a while :)

**Comment by ajotka** [comment]
@mapk I'm working on your proposal by my own (I don't want to promise anything yet!), but I found a bug. 
When you add header and/or footer, selecting don't work. If you select cell with index [0 0] it will also select header and footer with index [0 0].
I'm just reporting for keeping it in mind.

![image](https://user-images.githubusercontent.com/15144546/148776112-d372563f-37ae-4870-a239-4ae3492f019a.png)

**Comment by danielbachhuber** [comment]
Here's a PR for `colspan` support, if anyone would like to test it out: https://github.com/WordPress/gutenberg/pull/45981

**Comment by CloundNine** [comment]
The Getwid By MotoPress has a table block that does have merged cells. In case anyone comes across this and is looking for a quick fix.

**Comment by isuke01** [comment]
5 years later and still no support for merging from UI?! 😮
the https://github.com/WordPress/gutenberg/pull/45981 Is just in html editing, but I can't deliver something like that for the client.

**Comment by mot-K** [comment]
Is there something new here? I'm surprised this feature still doesn't exist after all this time.

**Comment by t-hamano** [comment]
Implementing the Merge/Unmerge functionality itself isn't difficult.

The hardest part is cell selection. It needs to be achievable with just the mouse or keyboard, so that any user can do it in any environment. We'll also need to handle screen reader announcements.
